src/test/java/seedu/address/commons/core/FunctionalBrowserTest.java

```diff
@@ -0,0 +1,53 @@
+package seedu.address.commons.core;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assumptions.assumeFalse;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.awt.GraphicsEnvironment;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+
+public class FunctionalBrowserTest {
+
+    @Test
+    public void getDesktop_sameFunctionalBrowserInstance_success() {
+        assumeFalse(GraphicsEnvironment.isHeadless(), "Test not ran on headless environments");
```

Comment:
See `FunctionalBrowser`: instead of skipping the testcase if its running on a headless environment, we can try to test the code except the point where the `desktop.browse` has to be called.

Reply:
Good point here!